simulator/infra/replacement/cache_replacement.cpp

```diff
@@ -68,10 +69,131 @@ std::size_t LRUCacheInfo::update()
     return lru_elem;
 }
 
+//////////////////////////////////////////////////////////////////
+
+struct node
+{
+    node(int new_value = 0, std::size_t new_pointer = 0)
```

Comment:
Please add `explicit` specifier here to prevent constructing a node from integer.